### PR TITLE
Mantis 18706 - Repeat message processing tries to copy an index

### PR DESCRIPTION
**Pull request by bramley**
Avoid trying to copy indices when copying a message.
Explicitly set the finishsending date when copying a message, otherwise eventually a repeated campaign will not be sent because finishsending is in the past.

### CODE REVIEW

File: public_html/lists/admin/connect.php [hunk 1]
```diff
@@ -1907,6 +1908,16 @@ function repeatMessage($msgid)
     unset($e['second']);
     setMessageData($newid, 'embargo', $e);
 
+    $finishSending = time() + DEFAULT_MESSAGEAGE;
+    $finish = array(
```
Comment (oliverklee) [inline public_html/lists/admin/connect.php]: I find both variable names very similar, and the different names don't make the differences clear. Proposal: finishTimestamp and finishDateParts.

What do you think?

File: public_html/lists/admin/connect.php [hunk 2]
```diff
@@ -1870,14 +1870,15 @@ function repeatMessage($msgid)
...
-    //  Do not copy columns that use default values or are explicitly set
+    //  Do not copy columns that use default values or are explicitly set, or indices
     $columnsToCopy = array_diff(
         array_keys($DBstruct['message']),
         array(
             'id', 'entered', 'modified', 'embargo', 'status', 'sent', 'processed', 'astext', 'ashtml',
             'astextandhtml', 'aspdf', 'astextandpdf', 'viewed', 'bouncecount', 'sendstart', 'uuid',
         )
     );
+    $columnsToCopy = preg_grep('/^index_/', $columnsToCopy, PREG_GREP_INVERT);
```
Comment (oliverklee) [inline public_html/lists/admin/connect.php]: I propose changing the old variable name in line 1874 to something different as it now does not represent the columns to copy anymore, but more columns.

Comment (bramley) [inline public_html/lists/admin/connect.php]: The variable has a "lifetime" of 10 lines, so it is not difficult to work out how it is being used.  I am happy the way it is.

File: public_html/lists/admin/connect.php (hunk 1 above)
Comment (bramley) [inline public_html/lists/admin/connect.php]: I think it is clear that $finish is an array of parts of dates. To understand why it is like that you need to look into setMessageData(). I'm happy with the way it is.

Comment (oliverklee) [inline public_html/lists/admin/connect.php]: I'm just trying to help your PR get into the code base in its best, clearest, most readable form. If you'd prefer to not get these suggestions, please let me know.

**Pull request merged**